schemas/registration.json

```diff
@@ -98,7 +100,8 @@
       "valueType": "string",
       "description": "Unique identifier of the sample, assigned by the data provider.",
       "restrictions": {
-        "required": true
+        "required": true,,
+        "regex": "[A-Za-z0-9\\-\\.]{1,64}"
```

Comment:
before settling down with this, maybe we can take a look at all current donor/specimen/sample submitter IDs in the ICGC portal to see what have been used

Reply:
ill ask dusan for a list

Reply:
@junjun-zhang it looks like the only other special character i can find is an underscore.  once id had a space, but i think christina does not want to allow that.

Reply:
Definitely no space allowed in IDs.